### PR TITLE
Feat(enhancement): Departure angle

### PR DESCRIPTION
**Pull request by RisingLeaf**
## Feature Details
Adds a definable "departure angle" per system. This angle specifies the maximum offset the ship can have from the direction to the destination system

## UI Screenshots

https://user-images.githubusercontent.com/85687254/211944766-3a174a67-12d0-4ae2-89ab-45c0352a2637.mov

## Usage Examples
```html
system
    "departure angle" [angle]
        [link] [angle]
        [jump] [angle]
```

you can define a different angle for jump drives and hyperdrives using the link/jump argument

angles in degrees, meaning from 0> to 180

an angle of zero means, that there is no departure angle

## Testing Done
see screen recording

## Performance Impact
Some extra calculations when jumping, should not be too much

## Note
This will be especially useful for ringworlds departure, where the AI navigates you out of the ringworld, but you still jump through it 😆

### CODE REVIEW

File: source/AI.cpp [hunk 1]
```diff
@@ -1990,15 +1993,41 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 		return;
 
 	bool isJump = (ship.JumpNavigation().GetCheapestJumpType(ship.GetTargetSystem()).first == JumpType::JUMP_DRIVE);
```
Comment (RisingLeaf) [inline source/AI.cpp:1995]:
```suggestion
	bool isJump = (ship.JumpNavigation().GetCheapestJumpType(ship.GetTargetSystem()).first == JumpType::JUMP_DRIVE);

```

**Comment by warp-core** [comment]
```
> system
>     "departure angle" [angle]
>         [link] [angle]
>         [jump] [angle]
```

Gonna need some more explanation of what the children mean/do.

> angles in degrees, meaning from 0 to 360

I'm not sure anything outside the range [0, 180] makes sense. Unless the angle actually corresponds to the size of the allowed arc, and not the difference between the current position vector and the target direction.

> This angle specifies the maximum offset the ship can have from the direction to the destination system

Suggests it is the latter, though.

**Comment by RisingLeaf** [comment]
> ```
> > system
> >     "departure angle" [angle]
> >         [link] [angle]
> >         [jump] [angle]
> ```
> 
> Gonna need some more explanation of what the children mean/do.

hmm, thought it would be obvious, one specifies when using hyperdrive/scramdrive the other one when using jump drive

> > angles in degrees, meaning from 0 to 360
> 
> I'm not sure anything outside the range [0, 180] makes sense. Unless the angle actually corresponds to the size of the allowed arc, and not the difference between the current position vector and the target direction.
> 
> > This angle specifies the maximum offset the ship can have from the direction to the destination system
> 
> Suggests it is the latter, though.

Correct

**Comment by RisingLeaf** [comment]
Also, only angles from 0 to 180 make sense yes, everything else gets handled the same, while negative angles make it impossible to jump out of the system.

**Comment by petervdmeer** [comment]
Is this PR related to https://github.com/endless-sky/rfcs/pull/1 ?

**Comment by RisingLeaf** [comment]
> Is this PR related to https://github.com/endless-sky/rfcs/pull/1 ?

I did not even know this exists, but now that you point on it, should I follow the proposed syntax?

**Comment by petervdmeer** [comment]
> I did not even know this exists, but now that you point on it, should I follow the proposed syntax?

Providing review-comments on the RFC (on why another syntax is better) is also an option.
But if the proposed syntax works for you, then I suggest we follow it here. (And if you review that RFC and don't have comments, then I'll also merge it.)

**Comment by RisingLeaf** [comment]
I left a review comment on the rfc

**Pull request closed without merging**